### PR TITLE
refactor: move env vars to json file

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Changes proposed in this pull request:

- all variables moved from .env to config.json
- default location of the config.json
- if desire to change that location, do so by overriding location with an env-var / .env file: `AICORE_HOME`
- deployment supports mounting secret as a volume

**Related issue(s)**

#197

### CODE REVIEW

File: src/utils/config.py [hunk 1]
```diff
@@ -56,12 +56,18 @@ def get_config() -> Config:
...
-        data = yaml.safe_load(f)
-    config = Config(**data)
-    return config
+    try:
+        with config_file.open() as file:
+            data = json.load(file)
+        # Extract only the "models" part of the configuration
+        models_data = data.get("models", [])
+        config = Config(models=models_data)
+        return config
```
Comment (muralov) [inline src/utils/config.py:70]:
```suggestion
        return config
     except json.JSONDecodeError as e:
       logging.error(f"Invalid JSON format in config file {config_path}: {e}")
       raise
```

File: src/utils/settings.py [hunk 2]
```diff
@@ -13,23 +14,49 @@ def is_running_pytest() -> bool:
...
+    if os.path.exists(config_path):
+        try:
+            # Load the configuration from the given path and set the environment variables.
+            with open(config_path) as file:
+                config_file = json.load(file)
+
+                # Set environment variables for all keys except "models"
+                for key, value in config_file.items():
+                    if key != "models":  # Skip models
+                        os.environ[key] = str(value)
```
Comment (muralov) [inline src/utils/settings.py:35]: please add the following error handling too:
```
     except json.JSONDecodeError as e:
       logging.error(f"Invalid JSON format in config file {config_path}: {e}")
       raise
```

**Review by muralov — CHANGES_REQUESTED** [review]

File: src/utils/config.py (hunk 1 above)
Comment (grischperl) [reply src/utils/config.py:70]: done

File: src/utils/settings.py (hunk 2 above)
Comment (grischperl) [reply src/utils/settings.py:35]: done

File: src/utils/settings.py [hunk 3]
```diff
@@ -13,23 +14,49 @@ def is_running_pytest() -> bool:
     return "pytest" in sys.modules
 
 
-project_root = os.path.dirname(os.path.abspath(__file__))
+def load_env_from_json() -> None:
+    """Load the configuration from the config.json file."""
+    config_path = os.getenv(
+        "AICORE_HOME",
+        os.path.abspath(
+            "../../go/src/github.com/kyma-project/kyma-companion/config/config.json"
```
[the remark this answers is not in the document]
Comment (grischperl) [reply src/utils/settings.py]: tested and done

File: tests/unit/utils/test_config.py
[the remark this answers is not in the document]
Comment (grischperl) [reply tests/unit/utils/test_config.py:1]: done

File: src/utils/settings.py [hunk 4]
```diff
@@ -13,23 +15,50 @@ def is_running_pytest() -> bool:
...
+                    if key != "models":  # Skip models
+                        os.environ[key] = str(value)
+        except json.JSONDecodeError as e:
+            logging.error(f"Invalid JSON format in config file {config_file}: {e}")
+            raise
+        except Exception as e:
+            logging.error(f"Error loading config from {config_path}: {e}")
+            raise
+    else:
+        logging.error(f"Config file not found at {config_path}.")
```
Comment (muralov) [inline src/utils/settings.py:41]: I think we need to raise an exception here. If a config file cannot be found, the app cannot start anyway.

File: src/utils/settings.py [hunk 5]
```diff
@@ -13,23 +15,50 @@ def is_running_pytest() -> bool:
...
 
 if is_running_pytest():
-    # Use .test.env for tests
+    # For tests use .env.test if available
     env_path = find_dotenv(".env.test")
     if env_path and os.path.exists(env_path):
         repository = RepositoryEnv(env_path)
         for key, value in repository.data.items():
             os.environ[key] = str(value)
         config = Config(repository)
```
Comment (muralov) [inline src/utils/settings.py:51]: I see we still use `.env.test` is this correct?

File: tests/unit/utils/test_config.py [hunk 6]
```diff
@@ -27,11 +38,19 @@
...
+                "models": [
+                    {
+                        "name": "single_model",
+                        "deployment_id": "single_dep",
+                        "temperature": 1
+                    }
+                ]
+            }
             """,
             Config(
```
Comment (muralov) [inline tests/unit/utils/test_config.py:55]: will add the `"VARIABLE_NAME": "value"` expected config too?

File: tests/unit/utils/test_config.py [hunk 7]
```diff
@@ -7,17 +7,28 @@
...
+                    {
+                        "name": "model2",
+                        "deployment_id": "dep2",
+                        "temperature": 0.5
+                    }
+                ]
+            }
             """,
             Config(
                 models=[
```
Comment (muralov) [inline tests/unit/utils/test_config.py:34]: please add the the other variables to expected config too,
                "VARIABLE_NAME": "value",
                "VARIABLE_NAME2": "value2",

File: src/utils/settings.py (hunk 4 above)
Comment (grischperl) [reply src/utils/settings.py:41]: done

File: tests/unit/utils/test_config.py (hunk 7 above)
Comment (grischperl) [reply tests/unit/utils/test_config.py:34]: This test expects only the model configuration to be returned. The env variables are set in `settings.py` function `load_env_from_json()`

File: tests/unit/utils/test_config.py (hunk 6 above)
Comment (grischperl) [reply tests/unit/utils/test_config.py:55]: added unit test for the env variables

File: tests/unit/utils/test_config.py (hunk 7 above)
Comment (grischperl) [reply tests/unit/utils/test_config.py:34]: added a unit test for `load_env_from_json()`

File: src/utils/settings.py (hunk 5 above)
Comment (grischperl) [reply src/utils/settings.py:52]: yes, however if no `.env.test` is found, then the config.json file is used

File: src/utils/settings.py (hunk 4 above)
Comment (muralov) [reply src/utils/settings.py:41]: we should make it work everywhere. I can help you with that.

**Review by muralov — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**